test/test_t2hahn.py

```diff
@@ -0,0 +1,133 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Test T2Hahn experiment
+"""
+import numpy as np
+
+from qiskit.utils import apply_prefix
+from qiskit.test import QiskitTestCase
+from qiskit_experiments.library.characterization.t2hahn import T2Hahn
+from qiskit_experiments.test.t2hahn_backend import T2HahnBackend
+
+
+class TestT2Hahn(QiskitTestCase):
+    """Test T2Hahn experiment"""
+
+    __tolerance__ = 0.1
+
+    def test_t2hahn_run_end2end(self):
+        """
+        Run the T2Hahn backend on all possible units
+        """
+        for unit in ["s", "ms", "us", "ns", "dt"]:
+            if unit in ("s", "dt"):
+                dt_factor = 1
+            else:
+                dt_factor = apply_prefix(1, unit)
+            osc_freq = 0.1 / dt_factor
+            estimated_t2hahn = 20
+            # Set up the circuits
+            qubit = 0
+            if unit == "dt":  # dt requires integer values for delay
+                delays = list(range(1, 46))
+            else:
+                delays = np.append(
+                    (np.linspace(1.0, 15.0, num=15)).astype(float),
+                    (np.linspace(16.0, 45.0, num=59)).astype(float),
+                )
+            exp = T2Hahn(qubit, delays, unit=unit)
+            default_p0 = {
+                "A": 0.5,
+                "T2": estimated_t2hahn,
+                "frequency": osc_freq,
```

Comment:
Frequency is not part of the user guess

Reply:
Deleted it, I didn't understand that this are "guesses" for the fitter.